src/bundler/mod.rs

```diff
@@ -118,17 +133,117 @@ impl<'a> Bundler<'a> {
                 tx.set_gas_price(max_fee_per_gas);
             }
         };
-        let res = client.send_transaction(tx, None).await?.await?;
+        let tx = client.send_transaction(tx, None).await?;
+        let tx_hash = tx.tx_hash();
+        debug!("Send bundles with transaction: {tx:?}");
 
-        debug!("Send bundles with ret: {res:?}");
-        Ok(())
+        let res = tx.await?;
+        debug!("Send bundles with receipt: {res:?}");
+        Ok(tx_hash)
     }
 }
 
+pub struct BundlerService {
+    pub bundlers: Vec<Bundler>,
+    pub bundle_interval: u64,
+    pub running: Arc<Mutex<bool>>,
+}
+
+fn is_running(running: Arc<Mutex<bool>>) -> bool {
+    let r = running.lock();
+    *r
+}
+
+impl BundlerService {
+    pub fn new(
+        wallet: Wallet,
+        beneficiary: Address,
+        uopool_grpc_client: UoPoolClient<tonic::transport::Channel>,
+        entry_points: Vec<Address>,
+        eth_client_address: String,
+        bundle_interval: u64,
+    ) -> Self {
+        let bundlers: Vec<Bundler> = entry_points
+            .iter()
+            .map(|entry_point| {
+                Bundler::new(
+                    wallet.clone(),
+                    beneficiary,
+                    uopool_grpc_client.clone(),
+                    *entry_point,
+                    eth_client_address.clone(),
+                )
+            })
+            .collect();
+
+        Self {
+            bundlers,
+            bundle_interval,
+            running: Arc::new(Mutex::new(false)),
+        }
+    }
+
+    pub async fn send_bundles_now(&self) -> anyhow::Result<H256> {
+        info!("Sending bundle now");
+        let mut tx_hashes: Vec<H256> = vec![];
+        for bundler in self.bundlers.iter() {
+            info!("Sending bundle for entry point: {:?}", bundler.entry_point);
+            let tx_hash = bundler.send_next_bundle().await?;
+            tx_hashes.push(tx_hash)
+        }
+
+        // FIXME: Because currently the bundler support multiple bundler and
+        // we don't have a way to know which bundler is the one that is
```

Comment:
@Vid201 Do you have any good ideas on addressing the multiple entry point here?

Reply:
Hmmm, I asked in Telegram about that.

Reply:
Send bundle on  each entry point.

Reply:
It is sending on each entry point. It is just the standard EIP output  only have one transaction hash result. So I would just return one result.

Reply:
https://github.com/eth-infinitism/account-abstraction/blob/develop/eip/EIPS/eip-4337.md#-debug_bundler_sendbundlenow

Reply:
Ohh, I get it now. The current tests are meant for only one entry point.

Let's make the function compatible with tests now (e.g., return the first element of vec of bundlers), and leave a comment in the code to fix later.